### PR TITLE
NEW Set many_many_extraFields data via the ORM

### PR DESCRIPTION
**Pull request by GuySartorelli**
There is currently no easy way to set extra fields on a `many_many` relationship using the ORM after items have already been added. This PR fills that gap.

The implementation closely follows the way these fields are set in `add()`.
I did try using an `SQLUpdate` but it didn't work with composite fields.

See silverstripe/developer-docs#10 for docs.

## Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10425

### CODE REVIEW

File: src/ORM/ManyManyList.php [hunk 1]
```diff
@@ -433,6 +433,58 @@ public function removeAll()
         }
     }
 
+    /**
+     * Set the extra field data for a single row of the relationship join
+     * table, given the known child ID.
+     *
+     * @param int $itemID The ID of the child for the relationship
+     * @param array $data The data to set, with field names as keys and values as values
+     * @throws BadMethodCallException if the foreign ID has not been set
```
Comment (kinglozzer) [inline src/ORM/ManyManyList.php:442]: I think this needs to be updated to `InvalidArgumentException`?

File: tests/php/ORM/ManyManyListTest.php [hunk 2]
```diff
@@ -73,14 +72,41 @@ public function testAddCompositedExtraFieldsWithSortColumn0()
...
+        $money->setCurrency('USD');
+
+        $obj->Clients()->add($obj2, [
+            'Worth' => $money,
+            'Reference' => 'Foo',
+        ]);
+        $obj->Clients()->setExtraData($obj2->ID, [
+            'Worth' => $money,
+            'Reference' => 'Foo',
+        ]);
```
Comment (kinglozzer) [inline tests/php/ORM/ManyManyListTest.php:102]: The call to `setExtraData()` sets the same data as `add()`, so I think this test would pass even if `setExtraData()` doesn’t have the desired effect - could you change the data passed to the `setExtraData()` call?

**Review by kinglozzer — CHANGES_REQUESTED** [review]
I think this is a worthwhile addition, just a couple of minor changes

File: tests/php/ORM/ManyManyListTest.php (hunk 2 above)
Comment (GuySartorelli) [reply tests/php/ORM/ManyManyListTest.php:102]: Yes, good point. Done.

File: src/ORM/ManyManyList.php (hunk 1 above)
Comment (GuySartorelli) [reply src/ORM/ManyManyList.php]: Yes, true. That was from my failed `SQLUpdate` implementation. I've updated the exception class and description here.

**Comment by GuySartorelli** [comment]
Rebased on 4 to get latest CI fixes

**Comment by GuySartorelli** [comment]
Not sure why that one unit test is failing, but it's not related to this PR.

**Review by kinglozzer — APPROVED** [review]

**Pull request merged**